pkg/subscriptions/subscriptions.js

```diff
@@ -0,0 +1,276 @@
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/* global jQuery   */
+/* global cockpit  */
+/* global po */
+
+define([
+    "jquery",
+    "latest/cockpit",
+    "latest/po",
+    "latest/mustache"
+], function($, cockpit, po, Mustache) {
+    "use strict";
+
+    cockpit.locale(po);
+
+    function debug() {
+        if (window.debugging == "all" || window.debugging == "subscriptions")
+            console.debug.apply(console, arguments);
+    }
+
+    var subscriptions = { };
+
+    PageSubscriptionsRegister.prototype = {
+        _init: function() {
+            this.id = "subscriptions-register-dialog";
+            var self = this;
+            var dlg = $('#subscriptions-register-dialog');
+            $('#account-register-start').on('click', function() {
+                self.register_system();
+                dlg.modal('hide');
+            });
+
+            dlg.on('show.bs.modal', function() {
+                self.enter();
+            });
+            dlg.on('shown.bs.modal', function() {
+                self.show();
+            });
+            dlg.on('hidden.bs.modal', function() {
+                self.leave();
+            });
+        },
+
+        show: function() {
+            $('#subscription-register-url').focus();
+        },
+
+        enter: function() {
+            $('#subscription-register-url').val("");
+            $('#subscription-register-username').val("");
+            $('#subscription-register-password').val("");
+        },
+
+        leave: function() {
+        },
+
+        register_system: function() {
+            var args = ['subscription-manager', 'register', '--auto-attach'];
+            var url = $('#subscription-register-url').val().trim();
+            var username = $('#subscription-register-username').val().trim();
+            var password = $('#subscription-register-password').val().trim();
+
+            if (url !== '')
+                args.push('--serverurl=' + url);
+            if (username !== '')
+                args.push('--username=' + username);
+            if (password !== '')
+                args.push('--password=' + password);
+
+            var process = cockpit.spawn(args, { directory: '/', superuser: true })
+                .done(function(output) {
+                    // parse for success
+                    var success = (output.indexOf('system has been registered') > -1);
```

Comment:
If you're going to be parsing output, you need to make sure to spawn with the ```LC_ALL=C``` environment variable to disable any localization.